src/routes/safe/store/actions/createTransaction.js

```diff
@@ -25,24 +25,31 @@ import {
 import { getErrorMessage } from '~/test/utils/ethereumErrors'
 import { ZERO_ADDRESS } from '~/logic/wallets/ethAddresses'
 import { SAFELIST_ADDRESS } from '~/routes/routes'
+import type { TransactionProps } from '~/routes/safe/store/models/transaction'
 
-const getLastPendingTxNonce = async (safeAddress: string): Promise<number> => {
-  let nonce
-
+const getLastTx = async (safeAddress: string): Promise<TransactionProps> => {
   try {
     const url = buildTxServiceUrl(safeAddress)
-
     const response = await axios.get(url, { params: { limit: 1 } })
-    const lastTx = response.data.results[0]
 
-    nonce = lastTx.nonce + 1
-  } catch (err) {
-    // use current's safe nonce as fallback
-    const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
-    nonce = (await safeInstance.nonce()).toString()
+    return response.data.results[0]
+  } catch (e) {
+    console.error('failed to retrieve last Tx from server', e)
+    return null
   }
+}
 
-  return nonce
+const getSafeNonce = async (safeAddress: string): Promise<string> => {
+  // use current's safe nonce as fallback
+  const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
+  return (await safeInstance.nonce()).toString()
+}
+
+const getNonce = async (txNonce, lastTx, safeAddress) => {
```

Comment:
Maybe we should give a little bit more context to this function, something like `getNewTxNonce` or add some comments why we did this

Reply:
great! changed its name.

I just moved to a function to make it more readable.